core/storage/email/gae_models_test.py

```diff
@@ -128,6 +149,168 @@ def test_get_by_hash_behavior_with_sent_datetime_lower_bound(self):
+        model = email_models.GeneralFeedbackEmailReplyToIdModel
+
+        created = model.create('user_id', 'thread_id')
+        created.put()
+
+        result = model.get_by_reply_to_id(created.reply_to_id)
+
+        result = model.get_by_reply_to_id('non_existent_reply_to_id')
+
+        self.assertEqual(result, None)
+
+    def test_get_can_be_used_to_retrive_model(self):
+        created = email_models.GeneralFeedbackEmailReplyToIdModel.create(
+            'user_id', 'thread_id')
+        created.put()
+
+        result = email_models.GeneralFeedbackEmailReplyToIdModel.get(
+            'user_id', 'thread_id', strict=False)
+
+        self.assertNotEqual(result, None)
+        self.assertIn('user_id.thread_id', str(result.key))
+
+    def test_get_returns_nothing_with_bad_id_if_not_strict(self):
+        created = email_models.GeneralFeedbackEmailReplyToIdModel.create(
+            'user_id', 'thread_id')
+        created.put()
+
+        result = email_models.GeneralFeedbackEmailReplyToIdModel.get(
+            'bad_user_id', 'bad_thread_id', strict=False)
+        self.assertEqual(result, None)
+
+        with self.assertRaises(Exception):
+            result = email_models.GeneralFeedbackEmailReplyToIdModel.get(
+                'bad_user_id', 'bad_thread_id')
+
+    def test_get_returns_throws_with_bad_id_if_strict(self):
+        created = email_models.GeneralFeedbackEmailReplyToIdModel.create(
+            'user_id', 'thread_id')
+        created.put()
+
+        with self.assertRaises(Exception):
+            email_models.GeneralFeedbackEmailReplyToIdModel.get(
+                'bad_user_id', 'bad_thread_id')
+
+    def test_get_multi_by_user_ids_works_correctly_with_multiple_ids(self):
+        model = email_models.GeneralFeedbackEmailReplyToIdModel
+
+        created1 = model.create('user_id_1', 'thread_id')
+        created1.put()
+        created2 = model.create('user_id_2', 'thread_id')
+        created2.put()
+
+        result = model.get_multi_by_user_ids(
+            ['user_id_1', 'user_id_2'], 'thread_id')
+
+        self.assertEqual(len(result), 2)
+        self.assertIn('user_id_1', result.keys())
+        self.assertIn('user_id_2', result.keys())
+        self.assertEqual(result['user_id_1'], created1)
+        self.assertEqual(result['user_id_2'], created2)
+
+    def test_get_multi_by_user_ids_works_correctly_with_one_id(self):
+        model = email_models.GeneralFeedbackEmailReplyToIdModel
+
+        created1 = model.create('user_id_1', 'thread_id')
+        created1.put()
+        created2 = model.create('user_id_2', 'thread_id')
+        created2.put()
+
+        result = model.get_multi_by_user_ids(
+            ['user_id_1'], 'thread_id')
+
+        self.assertEqual(len(result), 1)
+        self.assertIn('user_id_1', result.keys())
+        self.assertEqual(result['user_id_1'], created1)
+
+    def test_get_multi_by_user_ids_works_correctly_with_no_id(self):
```

Comment:
Here & above/below: can we be more specific than 'works correctly'? As an example, this test could be named something along the lines of:

``test_get_multi_with_no_user_ids_returns_empty_list``

Reply:
Done. Struggled a bit with the last one; since the result is that the key must be there, with None associated to it.